lib/dependabot/update_checkers/java_script/npm_and_yarn/subdependency_version_resolver.rb

```diff
@@ -24,34 +24,42 @@ def initialize(dependency:, credentials:, dependency_files:,
           end
 
           def latest_resolvable_version
-            # TODO: Update subdependencies for npm lockfiles
-            return if package_locks.any? || shrinkwraps.any?
-
             raise "Not a subdependency!" if dependency.requirements.any?
 
-            updated_lockiles = yarn_locks.map do |yarn_lock|
-              updated_content = update_subdependency_in_lockfile(yarn_lock)
-              updated_lockfile = yarn_lock.dup
+            lockfiles = [*package_locks, *shrinkwraps, *yarn_locks]
+
+            updated_lockfiles = lockfiles.map do |lockfile|
+              updated_content = update_subdependency_in_lockfile(lockfile)
+              updated_lockfile = lockfile.dup
               updated_lockfile.content = updated_content
               updated_lockfile
             end
 
-            version_from_updated_lockfiles(updated_lockiles)
+            version_from_updated_lockfiles(updated_lockfiles)
           end
 
           private
 
           attr_reader :dependency, :credentials, :dependency_files,
                       :ignored_versions
 
-          def update_subdependency_in_lockfile(yarn_lock)
+          def update_subdependency_in_lockfile(lockfile)
             SharedHelpers.in_a_temporary_directory do
               write_temporary_dependency_files
 
-              updated_files =
-                run_yarn_updater(path: Pathname.new(yarn_lock.name).dirname)
-
-              updated_files.fetch("yarn.lock")
+              updated_files = if lockfile.name.end_with?("yarn.lock")
+                                run_yarn_updater(
+                                  path: Pathname.new(lockfile.name).dirname,
+                                  lockfile_name: lockfile.name
+                                )
+                              else
+                                run_npm_updater(
+                                  path: Pathname.new(lockfile.name).dirname,
```

Comment:
Given that the path is derived from the name, why not just pass the full name and then get the path from it in the `run_npm_updater` methods (and likewise the `run_yarn_updater` one)?